thoth/storages/graph/postgres.py

```diff
@@ -2125,6 +2125,117 @@ def get_analyzer_documents_count_all(self) -> int:
+        self,
+        package_name: str,
+        package_version: str,
+        index_url: str,
+    ) -> bool:
+        """Check if Aggregate Security Indicators (SI) results exists for Python package version."""
+        with self._session_scope() as session:
+            query = self._construct_query_get_si_aggregated_python_package_version(
+                session=session,
+                package_name=package_name,
+                package_version=package_version,
+                index_url=index_url
+            )
+
+        return query.count() > 0
+
+    def get_si_aggregated_python_package_version(
+        self,
+        package_name: str,
+        package_version: str,
+        index_url: str,
+    ) -> Dict[str, int]:
+        """Get Aggregate Security Indicators (SI) results per Python package version.
+
+        Examples:
+        >>> from thoth.storages import GraphDatabase
+        >>> graph = GraphDatabase()
+        >>> graph.get_si_aggregated_python_package_version(
+            package_name='thoth-common',
+            package_version='0.10.0',
+            index_url='https://pypi.org/simple'
+        )
+            {
+                'severity_high_confidence_high': 0,
+                'severity_high_confidence_low': 0,
+                'severity_high_confidence_medium': 0,
+                'severity_high_confidence_undefined': 0,
+                'severity_low_confidence_high': 0,
+                'severity_low_confidence_low': 0,
+                'severity_low_confidence_medium': 0,
+                'severity_low_confidence_undefined': 0,
+                'severity_medium_confidence_high': 0,
+                'severity_medium_confidence_low': 0,
+                'severity_medium_confidence_medium': 0,
+                'severity_medium_confidence_undefined': 0,
+                'number_of_analyzed_files': 0,
+                'number_of_files_total': 0,
+                'number_of_files_with_severities': 0,
+                'number_of_filtered_files': 0,
+                'number_of_python_files': 39,
+                'number_of_lines_with_comments_in_python_files': 922,
+                'number_of_blank_lines_in_python_files': 2760,
+                'number_of_lines_with_code_in_python_files': 9509,
+                'total_number_of_files': 75,
+                'total_number_of_lines': 36856,
+                'total_number_of_lines_with_comments': 2895,
+                'total_number_of_blank_lines': 6737,
+                'total_number_of_lines_with_code': 27224
+            }
+        """
+        with self._session_scope() as session:
+            query = self._construct_query_get_si_aggregated_python_package_version(
+                session=session,
+                package_name=package_name,
+                package_version=package_version,
+                index_url=index_url
+            )
+
+            result = query.order_by(SecurityIndicatorAggregatedRun.datetime.desc()).first()
```

Comment:
The ordering can be an expensive operation. Do we need it here? Does `first` obtain the very last inserted? Or compare based on reverted ids?

Reply:
in theory we can have si for same package run multiple times and we want to retrieve only the latest one. The alternative is that we do an upsert if there is a more recent datetime and we update the date, so when we use the query, we should receive only one result per package name version and we do not need to sort. 

Regarding `first()` I think it will just give the first result from the query, so we are not sure is the latest inserted I guess.

Reply:
The thing you described is relevant also for other entries (e.g. solver queries, ...). We can run the same workflow multiple times and obtain results from the last one (in most cases).

OK, let's have this in and fine-tune perf later as needed.